jax/_src/numpy/lax_numpy.py

```diff
@@ -986,7 +986,11 @@ def heaviside(x1, x2):
 def hypot(x1, x2):
   _check_arraylike("hypot", x1, x2)
   x1, x2 = _promote_dtypes_inexact(x1, x2)
-  return lax.sqrt(x1*x1 + x2*x2)
+  x1 = lax.abs(x1)
+  x2 = lax.abs(x2)
+  x1, x2 = where(x1 > x2, x1, x2), where(x1 < x2, x1, x2)
+  yx = lax.div(x2, x1)
```

Comment:
This will produce `inf` if `x1` is exactly 0, which is why the CI build is failing.

You need to test and defend against that case. I suspect you need something like the following (aside: I think it may be slightly cleaner to use `max` and `min` rather than `select` or `where`):

```
x1 = lax.abs(x1)
x2 = lax.abs(x2)
x1, x2 = lax.max(x1, x2), lax.min(x1, x2)
return lax.select(x1 == 0, x1, x1 * lax.sqrt(1 + lax.square(lax.div(x2, x1)))
```

Note you should also verify that the gradient in your plots does something sensible at 0 as well: I think you might have omitted 0 before? I suspect the formulation above will produce NaN gradients at 0, so we will probably have to change it to:
```
x1 = lax.abs(x1)
x2 = lax.abs(x2)
x1, x2 = lax.max(x1, x2), lax.min(x1, x2)
return lax.select(x1 == 0, x1, x1 * lax.sqrt(1 + lax.square(lax.div(x2, lax.select(x1 == 0, ones_like(x1), x1))))
```

See https://github.com/google/jax/issues/1052 for an explanation of why the second one may be needed, but I caution you it's pretty complicated.